components/feature/pwa/src/main/java/mozilla/components/feature/pwa/feature/WebAppHideToolbarFeature.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa.feature
+
+import android.net.Uri
+import android.view.View
+import androidx.core.net.toUri
+import androidx.core.view.isGone
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+
+/**
+ * Hides a custom tab toolbar for Progressive Web Apps and Trusted Web Activities.
+ *
+ * When the [Session] is inside a trusted scope, the toolbar will be hidden.
+ * Once the [Session] navigates to another scope, the toolbar will be revealed.
+ *
+ * @param toolbar Toolbar to show or hide.
+ * @param sessionId ID of the custom tab session.
+ * @param trustedScopes Scopes to hide the toolbar at.
+ * Scopes correspond to [WebAppManifest.scope]. They can be a path (PWA) or just an origin (TWA).
+ */
+class WebAppHideToolbarFeature(
+    private val sessionManager: SessionManager,
+    private val toolbar: View,
+    private val sessionId: String,
+    private val trustedScopes: List<Uri>
+) : Session.Observer, LifecycleAwareFeature {
```

Comment:
nit: In general I think it's cleaner to implement the observer interfaces internally in a feature - like we do in some others, but not all unfortunately. The main reason for that is that it won't pollute the public API of the feature with observer methods that make no sense to the developer using the feature in an app and that no other code should call anyways.

Reply:
I'll keep that in mind in the future. In this case I think its fine since we only have one method from the session observer, and it represents the core functionality of the feature.

Reply:
But you will still inherit all the methods with default implementation :)